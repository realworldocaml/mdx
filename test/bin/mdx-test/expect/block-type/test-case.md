It is possible to explicitly state the type of a block using the
`block-type` label, working around language header and content based
inference which can sometime lead to troublesome error messages.

The following blocks use a volontarily misleading language header that would
normally lead to errors if we let MDX infer the type of block based on them.

<!-- $MDX type=toplevel -->
```sh
# 1 + 1;;
```

<!-- $MDX type=ocaml -->
```sh
let x = 2
```

<!-- $MDX type=cram -->
```ocaml
$ echo "boom"
```

The include block type is somewhat redundant with the `file=...` label as
so it is not tested here.
