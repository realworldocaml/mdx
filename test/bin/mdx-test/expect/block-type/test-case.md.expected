It is possible to explicitly state the type of a block using the
`block-type` label, working around language header and content based
inference which can sometime lead to troublesome error messages.

The following blocks use a volontarily misleading language header that would
normally lead to errors if we let MDX infer the type of block based on them.

```ocaml block-type=toplevel
# 1 + 1;;
- : int = 2
```

```ocaml block-type=ocaml
let x = 2
```

```sh block-type=cram
$ echo "boom"
boom
```

The include block type is somewhat redundant with the `file=...` label as
so it is not tested here.
